### PR TITLE
feat: New Module `avm/res/communication/email-service`

### PR DESCRIPTION
**Pull request by krbar**
## Description

New Module `avm/res/communication/email-service`.

Related Module Proposal: https://github.com/Azure/Azure-Verified-Modules/issues/749

## Adding a new module

- [x] A proposal has been submitted and approved.
- [ ] I have included "Closes #{module_proposal_issue_number}" in the PR description.
- [ ] I have run `brm validate` locally to verify the module files.
- [x] I have run deployment tests locally to ensure the module is deployable.

## Pipeline references

| Pipeline |
| - |
| [avm.res.communication.email-service](https://github.com/krbar/bicep-registry-modules/actions/workflows/avm.res.communication.email-service.yml) |

### CODE REVIEW

**Comment by AlexanderSehr** [comment]
@matebarabas & @jtracey93 , has this module been onboarded to the MCR?

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -21,6 +21,7 @@
 /avm/res/cache/redis/ @Azure/avm-res-cache-redis-module-owners-bicep @Azure/avm-core-team-technical-bicep
 /avm/res/cdn/profile/ @Azure/avm-res-cdn-profile-module-owners-bicep @Azure/avm-core-team-technical-bicep
 /avm/res/cognitive-services/account/ @Azure/avm-res-cognitiveservices-account-module-owners-bicep @Azure/avm-core-team-technical-bicep
+/avm/res/communication/email-service/ @Azure/avm-res-communication-emailservice-module-owners-bicep @Azure/avm-core-team-technical-bicep
```
Comment (eriqua) [inline .github/CODEOWNERS:24]: Placeholder. Please revolve after onboarding to MCR is complete

Comment (krbar) [reply .github/CODEOWNERS:24]: I just got an info from @matebarabas that the MCR onboarding is completed.

File: avm/res/communication/email-service/main.bicep [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    name: '${uniqueString(deployment().name, location)}-email-domain-${index}'
+    params: {
+      emailServiceName: email.name
+      name: domain.name
+      location: location
+      domainManagement: domain.?domainManagement
+      userEngagementTracking: domain.?userEngagementTracking
+      senderUsernames: domain.?senderUsernames
+      roleAssignments: domain.?roleAssignments
+      lock: domain.?lock
```
Comment (AlexanderSehr) [inline avm/res/communication/email-service/main.bicep:88]:
```suggestion
      lock: domain.?lock ?? lock
```

File: avm/res/communication/email-service/main.bicep [hunk 3]
```diff
@@ -0,0 +1,181 @@
+metadata name = 'Email Services'
+metadata description = 'This module deploys an Email Service'
+metadata owner = 'Azure/module-maintainers'
+
+@minLength(1)
+@maxLength(63)
+@description('Required. Name of the email service to create.')
+param name string
+
+@description('Optional. Location for all Resources.')
```
Comment (AlexanderSehr) [inline avm/res/communication/email-service/main.bicep:10]: Hey @krbar, given what you specified in the tests - should the default location always be 'global'?

Comment (krbar) [reply avm/res/communication/email-service/main.bicep:10]: Thanks, I changed it to "global"

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**